Fix DynamicFont breaking mouse grab in inspector spinners

opl- opened the pull request:
This was caused by DynamicFont not specifying which property was edited, resulting in the whole inspector being recreated

Fixes #25046

Edit: Valuable lessons about code formatting and the tools associated have been learned.

pouleyKetchoupp commented on scene/resources/dynamic_font.cpp:
```diff
@@ -795,18 +792,22 @@ int DynamicFont::get_spacing(int p_type) const {
 }
 
 void DynamicFont::set_spacing(int p_type, int p_value) {
+#define SET_SPACING_VALUE(NAME) \
+	spacing_##NAME = p_value;   \
+	emit_changed();             \
+	_change_notify("extra_spacing_" #NAME);
+
```
In this trivial case, the define is probably not needed. It only saves a few lines of code and the code would be more easily readable without it.

opl- replied:
I was a bit on the fence about this myself. The main reason for this was that seemingly `_change_notify` was always called before `emit_changed`, but looking at other resources it's not a requirement.

I'll push the change after testing it to make sure. It seems current master broke GDScript's `tool` which I'm relying on for testing.

akien-mga replied:
Even it `emit_changed()` should be kept first, it's ok to have it duplicated in the four branches.

opl- replied:
Changed it and rebased the branch to an older commit from `master` as #40598 doesn't yet have support for `tool` which is rather useful for testing here.

pouleyKetchoupp commented:
This fix would work, although we might also want to properly fix the more general unwanted behavior with spinners (probably outside of this PR's scope).
See https://github.com/godotengine/godot/issues/25046#issuecomment-663890179

akien-mga commented on scene/resources/dynamic_font.cpp:
```diff
@@ -677,7 +677,7 @@ DynamicFontAtSize::~DynamicFontAtSize() {
 /////////////////////////
 
-void DynamicFont::_reload_cache() {
+void DynamicFont::_reload_cache(const char *p_triggering_property) {
```
Any specific reason for `const char *` over `const String &`?

opl- replied:
Lack of my knowledge about project conventions. `_change_notify` takes in `const char *`, so I reused the type.

opl- committed:
Fix DynamicFont breaking mouse grab in inspector spinners

This was caused by DynamicFont not specifying which property was edited, resulting in the whole inspector property tree being recreated.

Fixes #25046

aaronfranke commented:
@opl- Is this still desired? If so, it needs to be [rebased](https://docs.godotengine.org/en/latest/community/contributing/pr_workflow.html#the-interactive-rebase) and tested on the latest master branch.

opl- commented:
@aaronfranke It seems that this is still an issue on master, but since #45879 the property change detection system works differently, so I'd have to look into it again and create an entirely new PR.

Meanwhile, this fix is still relevant to the 3.2 branch so we could just change the target branch for this PR.

Calinou commented:
This pull request should be reopened against the `3.x` branch (I'm not sure if we can change the base branch cleanly).

opl- commented:
The Edit button next to the title shows a dropdown that allows changing the target branch. We could try to see if it works. If it doesn't, at least we learned something and I'll open a new PR.

Calinou commented:
@opl- It looks like it's not working, so I had to revert it. Please open a new pull request against the `3.x` branch.
